File: Basics_with_JavaScript/1112_While_Loop/12_While_loop_Excercise/06._Cake.js
function vladi(array){
    let wide = Number(array[0]);
    let length = Number(array[1]);
    let totalPeaces = wide * length;
    let takenPeaces = 0;
    isNoPeaces = false;

    
    let index = 2;
    let command = (array[index]);
    index++;

    while (command !== "STOP") {
        let take = Number(command);
        takenPeaces += take;
        if (takenPeaces > totalPeaces) {
            isNoPeaces = true;
            break
        }
        
        command = array[index];
        index++;
    }
// if command is STOP, or  taken peaces > total
    let diff = Math.abs(totalPeaces - takenPeaces);

    if (isNoPeaces) {
        console.log(`No more cake left! You need ${diff} pieces more.`);
    } else {
        console.log(`${diff} pieces are left.`);
    }
    
    }
   
    vladi(["10",
    "10",
    "20",
    "20",
    "20",
    "20",
    "21"])
    ;
    vladi(["10",
    "2",
    "2",
    "4",
    "6",
    "STOP"])
    ;

File: Basics_with_Python/PythonBasicCourse/5_While_Loop/12_While_Loop_Exercise/06._Cake.py
wide = int(input())
length = int(input())
total_peaces = wide * length
taken_peaces = 0

while taken_peaces != "STOP":
    taken_peaces = int(taken_peaces)
    total_peaces -= taken_peaces
    if total_peaces <= 0:
        break
    else:
        taken_peaces = input()

if taken_peaces == "STOP" and total_peaces >= 0:
    print(f"{total_peaces} pieces are left.")
elif total_peaces < 0:
    print(f"No more cake left! You need {abs(total_peaces)} pieces more.")

# wide = int(input())
# length = int(input())
# total_peaces = wide * length
# taken_peaces = 0
#
# while total_peaces > 0:
#     if taken_peaces == "STOP":
#         break
#     else:
#         taken_peaces = int(taken_peaces)
#         total_peaces -= taken_peaces
#         if total_peaces <= 0:
#             break
#         taken_peaces = input()
#
#
# if taken_peaces == "STOP" and total_peaces >= 0:
#     print(f"{total_peaces} pieces are left.")
# elif total_peaces < 0:
#     print(f"No more cake left! You need {abs(total_peaces)} pieces more.")
